--- rec-taskqueue_orig.cc	2021-10-03 09:47:30.010000000 +0200
+++ rec-taskqueue.cc	2021-10-03 10:17:26.083066971 +0200
@@ -30,10 +30,10 @@
 static pdns::stat_t s_almost_expired_tasks_run;
 static pdns::stat_t s_almost_expired_tasks_exceptions;
 
-static void resolve(const struct timeval& now, bool logErrors, const pdns::ResolveTask& task)
+static void resolve(const struct pdns::timeval& now, bool logErrors, const pdns::ResolveTask& task)
 {
   const string msg = "Exception while running a background ResolveTask";
-  SyncRes sr(now);
+  SyncRes sr(reinterpret_cast<const struct timeval&>(now));
   vector<DNSRecord> ret;
   sr.setRefreshAlmostExpired(task.d_refreshMode);
   try {
